### PR TITLE
feat(csslsrs): get document symbols

### PR DESCRIPTION
**Pull request by goulvenclech**
## What does this change?

Added `get_document_symbols` in CSSlsrs language service to extract document symbols from a given CSS document. (`crates/csslsrs/src/features/document_symbols.rs`)

## How is it tested?

Added tests for the document symbols extraction feature, including tests for class selectors, ID selectors, nested selectors, and at-rules. (`crates/csslsrs/tests/document_symbols.rs`)

## How is it documented?

For now, only in the function signature.

### CODE REVIEW

File: crates/weblsp/src/server.rs [hunk 1]
```diff
@@ -4,6 +4,7 @@ pub(crate) fn get_server_capabilities() -> serde_json::Value {
     let capabilities = ServerCapabilities {
         hover_provider: Some(lsp_types::HoverProviderCapability::Simple(true)),
         color_provider: Some(lsp_types::ColorProviderCapability::Simple(true)),
+        document_symbol_provider: Some(lsp_types::OneOf::Left(true)),
```
[the remark this answers is not in the document]
Comment (Princesseuh) [reply crates/weblsp/src/server.rs:7]: yea ok, types are a bit wonky here but that's what we support

File: crates/csslsrs/src/features/document_symbols.rs [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+                        at_rule
+                            .first_child()
+                            .filter(|child| child.kind() == CssSyntaxKind::CSS_MEDIA_QUERY_LIST)
+                            .map(|child| child.text_trimmed().to_string()),
+                        SymbolKind::NAMESPACE,
+                        range(line_index, child.text_trimmed_range(), encoding).unwrap(),
+                        Range::new(
+                            position(
+                                line_index,
+                                token.text_trimmed_range().start() - TextSize::from(1),
```
Comment (Princesseuh) [inline crates/csslsrs/src/features/document_symbols.rs:39]: what's this minus one thing?

File: crates/csslsrs/tests/document_symbols.rs [hunk 3]
```diff
@@ -0,0 +1,608 @@
+#![allow(deprecated)]
```
Comment (Princesseuh) [inline crates/csslsrs/tests/document_symbols.rs:1]: would love a test for a at rule with nested properties like 

```css
@font-palette-values --my-palette {
  font-family: Bixa;
  base-palette: 1;
  override-colors: 0 #ff0000;
}
```

File: packages/language-server-tests-benchmarks/src/benchmarks/css/document_symbols.bench.ts [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+describe("DocumentSymbols", async () => {
+	bench("weblsp - DocumentSymbols", async () => {
+		await weblsp.sendDocumentSymbolRequest(weblspUri);
+	});
+
+	if (!process.env.CODSPEED) {
+		bench("vscode-css-languageserver - DocumentSymbols", async () => {
+			await vscodeLsp.sendDocumentSymbolRequest(vscodeLspUri);
+		});
+	}
```
Comment (Princesseuh) [inline packages/language-server-tests-benchmarks/src/benchmarks/css/document_symbols.bench.ts:24]:
```suggestion
describe("Document Symbols", async () => {
	bench("weblsp - Document Symbols", async () => {
		await weblsp.sendDocumentSymbolRequest(weblspUri);
	});

	if (!process.env.CODSPEED) {
		bench("vscode-css-languageserver - Document Symbols", async () => {
			await vscodeLsp.sendDocumentSymbolRequest(vscodeLspUri);
		});
	}
```
names here are human readable

File: crates/csslsrs/tests/document_symbols.rs (hunk 3 above)
Comment (goulvenclech) [reply crates/csslsrs/tests/document_symbols.rs:1]: Added. Tell me if the result suit you.

File: packages/language-server-tests-benchmarks/src/benchmarks/css/document_symbols.bench.ts [hunk 5]
```diff
@@ -0,0 +1,32 @@
+import { afterAll, bench, describe } from "vitest";
+import { startLanguageServer } from "../../server";
+import { fileURLToPath } from "url";
+
+const filePath = fileURLToPath(
+	new URL("../../../fixture/document_symbols_benchmark.css", import.meta.url)
```
Comment (Princesseuh) [inline packages/language-server-tests-benchmarks/src/benchmarks/css/document_symbols.bench.ts:6]:
```suggestion
	new URL("../../../fixture/document_symbols.css", import.meta.url)
```

**Review by Princesseuh — APPROVED** [review]
awesome work

**Pull request merged**